alternator/executor.cc

```diff
@@ -2397,6 +2404,30 @@ static future<executor::request_return_type> rmw_operation_return(rjson::value&&
     return make_ready_future<executor::request_return_type>(make_jsonable(std::move(ret)));
 }
 
+static future<std::unique_ptr<rjson::value>> get_previous_item(
```

Comment:
Please consider splitting the changes to get_previous_item() to a separate patch. It was really hard for me to understand what I was seeing.

If you really need to have two functions with the same name get_previous_item (I didn't yet understand why), please add comments explaining when to use each and how they differ. 

By the way, I noticed now in executor.cc (before your patch) that we have a _wrong_ forward deceleration of get_previous_item (that should just be removed). How awkward.